### PR TITLE
feat: new shellcheck linter

### PR DESCRIPTION
**Pull request by wwcchh0123**

### CODE REVIEW

**Comment by wwcchh0123** [comment]
The effects of the shellcheck can be found in the following PR:https://github.com/wwcchh0123/testrepo/pull/6

File: internal/linters/shell/shellcheck/shellcheck.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+var lintName = "shellcheck"
+
+func init() {
+	linters.RegisterPullRequestHandler(lintName, shellcheckHandler)
+}
+
+func shellcheckHandler(log *xlog.Logger, a linters.Agent) error {
+	/*
+	 * Shellcheck can not support check directories recursively, it is usually used in conjunction with the "find" command
```
Comment (CarlJi) [inline internal/linters/shell/shellcheck/shellcheck.go:22]: if so， we can just check the shell files in PR, Refer: https://github.com/qiniu/reviewbot/blob/master/internal/linters/linters.go#L98

File: internal/linters/shell/shellcheck/shellcheck.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	}
+	return b, nil
+}
+
+func (g *Shellcheck) Parse(log *xlog.Logger, output []byte) (map[string][]linters.LinterOutput, error) {
+	log.Infof("shellcheck output is being parsed")
+	return formatShellcheckOutput(output)
+}
+
+func formatShellcheckOutput(output []byte) (map[string][]linters.LinterOutput, error) {
```
[the remark this answers is not in the document]
Comment (wwcchh0123) [reply internal/linters/shell/shellcheck/shellcheck.go:117]: updated in line 116

File: internal/linters/shell/shellcheck/shellcheck.go (hunk 1 above)
Comment (wwcchh0123) [reply internal/linters/shell/shellcheck/shellcheck.go:22]: updated in line 67 to 75

**Pull request closed without merging**